internal/handlers/common/update.go

```diff
@@ -16,16 +16,34 @@ package common
 
 import (
 	"fmt"
+	"math/rand"
 
 	"github.com/FerretDB/FerretDB/internal/types"
 	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 )
 
 // UpdateDocument updates the given document with a series of update operators.
-func UpdateDocument(doc, update *types.Document) error {
-	for _, updateOp := range update.Keys() {
-		updateV := must.NotFail(update.Get(updateOp))
+func UpdateDocument(doc *types.Document, update any) error {
+	var updateDoc *types.Document
+	var isUpdateArray bool
+	for {
```

Comment:
Sorry what for here is an endless `for`? `update` is not changing.

Reply:
Yep, I overdid it just a bit. :smile: 
Good catch. Thanks! Fixed